geode-gfsh/src/integrationTest/java/org/apache/geode/management/internal/cli/commands/StartMemberUtilsTest.java

```diff
@@ -172,9 +177,14 @@ public void testAddMaxHeap() {
 
     // Only Max Heap Option Set
     StartMemberUtils.addMaxHeap(baseCommandLine, "32g");
-    assertThat(baseCommandLine.size()).isEqualTo(3);
-    assertThat(baseCommandLine).containsExactly("-Xmx32g", "-XX:+UseConcMarkSweepGC",
-        "-XX:CMSInitiatingOccupancyFraction=" + StartMemberUtils.CMS_INITIAL_OCCUPANCY_FRACTION);
+    if (getJavaMajorVersion() < 14) {
+      assertThat(baseCommandLine.size()).isEqualTo(3);
+      assertThat(baseCommandLine).containsExactly("-Xmx32g", "-XX:+UseConcMarkSweepGC",
+          "-XX:CMSInitiatingOccupancyFraction=" + StartMemberUtils.CMS_INITIAL_OCCUPANCY_FRACTION);
+    } else {
+      assertThat(baseCommandLine.size()).isEqualTo(1);
+      assertThat(baseCommandLine).containsExactly("-Xmx32g");
+    }
 
```

Comment:
Take a look at JUnit 5's `@EnabledOnJre(…)` and `@EnabledForJreRange(min=…, max=…)` annotations. There are also `@Disabled…` ones.

Reply:
You should change the assertion to use `hasSize`. You can even combine both if you want or even eliminate `hasSize` altogether since the other assertion will still fail if it's zero:
```
assertThat(baseCommandLine)
    .hasSize(1)
    .containsExactly("-Xmx32g");
```